trade-finance-manager-ui/server/api.js

```diff
@@ -68,28 +69,31 @@ const getFacilities = async (token, searchString = '') => {
   }
 };
 
+/**
+ * Makes a request to the GET /deals TFM API endpoint
+ * and throws an error if the page number is out of bounds
+ * @param {Object} queryParams Query parameters
+ * @param {string} token Authorisation token
+ * @returns {Object} Deals data and pagination metadata
+ * @throws {PageOutOfBoundsError} Will throw if the requested page number exceeds the maximum page number
+ */
 const getDeals = async (queryParams, token) => {
-  try {
-    const response = await axios({
-      method: 'get',
-      url: `${TFM_API_URL}/v1/deals`,
-      headers: generateHeaders(token),
-      params: queryParams,
-    });
-    if (response.data) {
-      return {
-        deals: response.data.deals,
-        count: response.data.pagination.totalItems,
-      };
-    }
-    return {
-      deals: [],
-      count: 0,
-    };
-  } catch (error) {
-    console.error('Unable to get deals %O', error);
-    return {};
+  const response = await axios({
+    method: 'get',
+    url: `${TFM_API_URL}/v1/deals`,
+    headers: generateHeaders(token),
+    params: queryParams,
+  });
+  const { deals, pagination } = response.data;
+
+  if (queryParams.page >= pagination?.totalPages && !(queryParams.page === 0 && pagination?.totalPages === 0)) {
```

Comment:
Found this a little hard to understand, could this just be the following? Not sure why we'd need a check for `!(page === 0 && totalPages === 0)`?

```js
if (page > pagination.totalPages) {
  // throw 
}
```

Should `totalPages` always be defined, save the `pagination?...`?

Reply:
Good spot - the reason for this was that the backend returned totalPages = 0 when there were no deals, but I realised it was better to remove this logic here and instead change the backend to return totalPages = 1 when there are no deals. You are also right, pagination is always defined, so there is no need for the ?